hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/command/index/TestExpressionIndex.scala

```diff
@@ -846,11 +852,122 @@ class TestExpressionIndex extends HoodieSparkSqlTestBase {
           var literal = Literal.create("rider-c")
           var dataFilter = EqualTo(lowerExpr, literal)
           verifyFilePruning(opts, dataFilter, metaClient, isDataSkippingExpected = true)
+          spark.sql(s"drop index idx_rider on $tableName")
 
           val fromUnixTime = resolveExpr(spark, unapply(functions.from_unixtime(functions.col("ts"), "yyyy-MM-dd")).get, tableSchema)
           literal = Literal.create("2023-11-07")
           dataFilter = EqualTo(fromUnixTime, literal)
           verifyFilePruning(opts, dataFilter, metaClient, isDataSkippingExpected = true)
+          spark.sql(s"drop index idx_datestr on $tableName")
```

Comment:
Can we add a separate test? I know the test time will increase a little bit due to creating the table again, but it's cleaner. If needed, we can split this test class into two so that they can run in parallel.